torch/csrc/dynamo/eval_frame.c

```diff
@@ -304,10 +305,100 @@ typedef struct cache_entry {
   struct cache_entry* next;
 } CacheEntry;
 
+#define Py_NONE_CACHE_ENTRY (CacheEntry*)Py_None
+
+static void destroy_cache_entry(CacheEntry* e);
+
+#define DECLARE_CACHE_ENTRY_ATTR(name) \
+static PyObject* CacheEntry_##name(CacheEntry* self, PyObject* _noargs) { \
+  PyObject* res = (PyObject*)self->name; \
+  Py_INCREF(res); \
+  return res; \
+}
+
+DECLARE_CACHE_ENTRY_ATTR(check_fn)
+DECLARE_CACHE_ENTRY_ATTR(code)
+DECLARE_CACHE_ENTRY_ATTR(next)
+
+static struct PyGetSetDef CacheEntry_properties[] = {
+    {"check_fn", (getter)CacheEntry_check_fn, NULL, NULL, NULL},
+    {"code", (getter)CacheEntry_code, NULL, NULL, NULL},
+    {"next", (getter)CacheEntry_next, NULL, NULL, NULL},
+    {NULL}};
+
+
+static PyObject* new_cache_entry(PyTypeObject* type, PyObject* args, PyObject* kwargs) {
+  CacheEntry *self;
+  self = (CacheEntry*) type->tp_alloc(type, 0);
+  if (self != NULL) {
+    // The corresponding decrefs for Py_None are in init_cache_entry.
+    Py_INCREF(Py_None);
+    self->check_fn = Py_None;
+    Py_INCREF(Py_None);
+    self->code = (PyCodeObject*)Py_None;
+    Py_INCREF(Py_None);
```

Comment:
This should be Py_NONE_CACHE_ENTRY no? ;) (TBH, it's probably clearer to just cast Py_None to the appropriate type as necessary, instead of macro'ing it)